chronos/asyncloop.nim

```diff
@@ -1069,15 +1068,21 @@ proc wait*[T](fut: Future[T], timeout = -1): Future[T] {.
   else:
     wait(fut, timeout.milliseconds())
 
+{.pop.}
+
 include asyncmacro2
 
-proc runForever*() =
+{.push raises: [Defect].}
+
+proc runForever*() {.raises: [Defect, CatchableError].} =
   ## Begins a never ending global dispatcher poll loop.
+  ## Raises different exceptions depending on the platform.
   while true:
     poll()
 
-proc waitFor*[T](fut: Future[T]): T =
+proc waitFor*[T](fut: Future[T]): T {.raises: [Defect, CatchableError].} =
```

Comment:
I think this procedure should not leak `CatchableError`.

Reply:
it raises the `CatchableError` that is potentially inside the future - it would have to return the future (or a `Result`) in order to only raise when `poll` raises